cluster/test/Cardano/Cluster/Environment/Spec.hs

```diff
@@ -53,67 +53,46 @@ prop_generatedEnvironmentIsValid (Cluster nodes) mport (SeparatedBy stateDir) =
-                , prop_environmentEdge env ((wPort, Just $ wPort + 100), (wPort +100, Nothing))
+                , conjoinWithContext "prop_edgeEnvironment"
+                    [ prop_noEnvVar   env "LISTEN"
+                    , prop_portWithin env "NODE_API_ADDRESS" (wPort, Nothing)
+                    , prop_portWithin env "NODE_DOC_ADDRESS" (wPort + 100, Nothing)
+                    ]
                 ]
   where
     prop_commonEnvironment env = conjoinWithContext "prop_commonEnvironment"
         [ prop_hasEnvVar  env "CONFIGURATION_FILE"
         , prop_hasEnvVar  env "CONFIGURATION_KEY"
         , prop_hasEnvVar  env "DB_PATH"
-        , prop_hasEnvVar  env "NODE_ID"
         , prop_hasEnvVar  env "LOG_CONFIG"
-        , prop_noEnvVar   env "LOG_SEVERITY"
-        , prop_hasEnvVarP env "REBUILD_DB" (`elem` ["True", "False"])
-        ]
-
-    prop_environmentCore env portRange = conjoinWithContext "prop_environmentCore"
-        [ prop_portWithin env "LISTEN" portRange
-        , prop_noEnvVar   env "NODE_API_ADDRESS"
-        , prop_noEnvVar   env "NODE_DOC_ADDRESS"
-        , prop_noEnvVar   env "WALLET_ADDRESS"
-        , prop_noEnvVar   env "WALLET_DOC_ADDRESS"
-        , prop_noEnvVar   env "WALLET_DB_PATH"
-        , prop_noEnvVar   env "WALLET_REBUILD_DB"
-        , prop_noEnvVar   env "NO_CLIENT_AUTH"
-        , prop_noEnvVar   env "TLSCERT"
-        , prop_noEnvVar   env "TLSKEY"
-        , prop_noEnvVar   env "TLSCA"
-        ]
-
-    prop_environmentRelay env portRange = conjoinWithContext "prop_environmentRelay"
-        [ prop_portWithin env "LISTEN" portRange
-        , prop_noEnvVar   env "NODE_API_ADDRESS"
-        , prop_noEnvVar   env "NODE_DOC_ADDRESS"
-        , prop_noEnvVar   env "WALLET_ADDRESS"
-        , prop_noEnvVar   env "WALLET_DOC_ADDRESS"
-        , prop_noEnvVar   env "WALLET_DB_PATH"
-        , prop_noEnvVar   env "WALLET_REBUILD_DB"
-        , prop_noEnvVar   env "NO_CLIENT_AUTH"
-        , prop_noEnvVar   env "TLSCERT"
-        , prop_noEnvVar   env "TLSKEY"
-        , prop_noEnvVar   env "TLSCA"
-        ]
-
-    prop_environmentEdge env (apiRange, docRange) = conjoinWithContext "prop_environmentEdge"
-        [ prop_portWithin env "NODE_API_ADDRESS" apiRange
-        , prop_portWithin env "NODE_DOC_ADDRESS" docRange
+        , prop_hasEnvVar  env "NODE_ID"
+        , prop_hasEnvVar  env "TLSCA"
         , prop_hasEnvVar  env "TLSCERT"
         , prop_hasEnvVar  env "TLSKEY"
-        , prop_hasEnvVar  env "TLSCA"
+        , prop_hasEnvVarP env "REBUILD_DB" (`elem` ["True", "False"])
         , prop_hasEnvVarP env "NO_CLIENT_AUTH" (`elem` ["True", "False"])
+        , prop_noEnvVar   env "LOG_SEVERITY"
         , prop_noEnvVar   env "WALLET_ADDRESS"
-        , prop_noEnvVar   env "WALLET_DOC_ADDRESS"
         , prop_noEnvVar   env "WALLET_DB_PATH"
+        , prop_noEnvVar   env "WALLET_DOC_ADDRESS"
         , prop_noEnvVar   env "WALLET_REBUILD_DB"
         ]
```

Comment:
Re-organized properties a bit since a lot can now be conflated. They share almost the same environment. There are just some "implicit" rules about the address ports we want to verify (in particular, relay nodes listen to a port that is +100 above core nodes...)